bilby/gw/likelihood.py

```diff
@@ -1531,76 +1534,53 @@ def log_likelihood_ratio(self):
-            per_detector_snr = self.calculate_snrs_relative_binning(waveform_ratio[interferometer.name], interferometer)
-
+            per_detector_snr = self.calculate_snrs_relative_binning(
+                waveform_ratio[interferometer.name], interferometer)
             d_inner_h += per_detector_snr.d_inner_h
-            optimal_snr_squared += np.real(
-                per_detector_snr.optimal_snr_squared)
+            optimal_snr_squared += np.real(per_detector_snr.optimal_snr_squared)
             complex_matched_filter_snr += per_detector_snr.complex_matched_filter_snr
 
         log_l = np.real(d_inner_h) - optimal_snr_squared / 2
         return float(log_l.real)
 
-    def find_maximum_likelihood_waveform(self, initial_parameter_guess,
-                                         parameter_bounds, iterations=10,
-                                         likelihood_threshold=1):
-        prev_log_likelihood = -np.inf
+    def find_maximum_likelihood_parameters(self, parameter_bounds,
+                                           iterations=1, atol=1e-3, maxiter=100):
+
+        parameter_bounds_list = self.get_parameter_list_from_dictionary(parameter_bounds)
+
         for i in range(iterations):
-            print("iter: %s" % i)
-            print("length of parameter_bounds", len(parameter_bounds))
-            log_likelihood = self.get_best_fit_parameters(
-                self.get_parameter_list_from_dictionary(parameter_bounds),
-                atol=1e-10, maxiter=10)  # change back to 500, no input
-            print("likelihood: %s" % log_likelihood)
-
-            self.maximum_likelihood_parameters.update(self.get_sky_frame_parameters())
-            self.set_fiducial_waveforms(self.maximum_likelihood_parameters)
+            logger.info("Optimizing fiducial parameters. Iteration : {}".format(i + 1))
+            output = differential_evolution(self.lnlike_scipy_maximize,
+                                            bounds=parameter_bounds_list, atol=atol, maxiter=maxiter)
+            updated_parameters_list = output['x']
+            updated_parameters = self.get_parameter_dictionary_from_list(updated_parameters_list)
+            self.set_fiducial_waveforms(updated_parameters)
+            self.setup_bins()
             self.compute_summary_data()
 
-            if np.abs(log_likelihood - prev_log_likelihood) < (
-                    likelihood_threshold):
-                print('Likelihood change threshold reached. Stopping.')
-                return
+        logger.info("Fiducial waveforms updated")
+        logger.info("Summary Data updated")
+        return updated_parameters
 
-            prev_log_likelihood = log_likelihood
-
-        print("Max iteration reached. Stopping.")
-        return
+    def lnlike_scipy_maximize(self, parameter_list):
+        self.parameters = self.get_parameter_dictionary_from_list(parameter_list)
+        return -self.log_likelihood_ratio()
```

Comment:
If `differential_evolution` is a minimization algorithm, shouldn't `lnlike_scipy_maximize` return `self.log_likelihood_ratio`? Also, remind me again: do our current maxL points have negative likelihood or positive?

Reply:
Our current maxL points have negative likelihood values but it should be positive.

Reply:
How are you confident that maxL should be posiitve? I am not too sure that is the case. It is a convention after all.

Reply:
I just think that should be the case because it's positive in both Greg's run and the paper. Maybe we should change our settings to the whatever Greg's run had? That way we'll have something to compare with